### PR TITLE
Dedupe advertisement collisions

### PR DESCRIPTION
**Pull request by marceloneil**

### CODE REVIEW

File: go/chat/bots/commands.go [hunk 1]
```diff
@@ -343,20 +344,25 @@ func (b *CachingBotCommandManager) ListCommands(ctx context.Context, convID chat
...
-			if !cmdDedup[key] {
-				res = append(res, cmd.ToOutput(ad.Username))
-				cmdDedup[key] = true
+			if typ, ok := cmdDedup[key]; !ok || chat1.AdvertRanking[ad.Typ] > chat1.AdvertRanking[typ] {
+				cmdOutputs[key] = cmd.ToOutput(ad.Username)
+				cmdDedup[key] = ad.Typ
 			}
 		}
 	}
+	for _, cmd := range cmdOutputs {
```
Comment (joshblum) [inline go/chat/bots/commands.go:362]: can you pre-allocate `res` since we know the size of `cmdOuputs`?

File: go/chat/bots/commands.go [hunk 2]
```diff
@@ -343,20 +344,25 @@ func (b *CachingBotCommandManager) ListCommands(ctx context.Context, convID chat
...
 		ad.Username = libkb.NewNormalizedUsername(ad.Username).String()
 		if ad.Advertisement.Alias != nil {
 			alias[ad.Username] = *ad.Advertisement.Alias
 		}
 		for _, cmd := range ad.Advertisement.Commands {
 			key := cmd.Name + ad.Username
-			if !cmdDedup[key] {
-				res = append(res, cmd.ToOutput(ad.Username))
-				cmdDedup[key] = true
+			if typ, ok := cmdDedup[key]; !ok || chat1.AdvertRanking[ad.Typ] > chat1.AdvertRanking[typ] {
```
Comment (joshblum) [inline go/chat/bots/commands.go:356]: can't we just compare the types directly? they are just an enum

Comment (marceloneil) [reply go/chat/bots/commands.go:356]: Yeah, my only concern is if we add new advertisement types they won't necessarily have their 'rank' the same as the enum value. We could worry about that later though, lmk what you think

Comment (joshblum) [reply go/chat/bots/commands.go:356]: we can reintroduce this if we add a new type, can you drop a comment in the avdl? old clients won't handle the value missing from the map anyway

**Pull request merged**